### DISCUSSION TITLE
Maybe time to start using a dev branch and proper merge to production patterns?

### DISCUSSION THREAD

**Discussion by marcsyp** [Mikubill/sd-webui-controlnet#604] [Ideas]
There has been a whole lot of activity lately and a lot of breaking changes to the codebase.  I've seen the discussion around implementing tests, which is great, but I still see what appears to be a good number of contributors pushing directly to the main branch, which is a pretty dangerous practice for a codebase in production.  

With a growing number of features and fixes being worked on, does it make sense to start using feature branches and/or a dev branch so that pull requests can be merged and tested before merging to the production main branch?  Perhaps I don't have visibility into the whole process behind the scenes -- not trying to rock the boat here, but I noticed that many users have expressed some frustration with the speed at which things are changing/breaking/fixing, and I for one am kind of afraid to pull the latest version based on what I've been seeing.  I'm still on 4c13542 from a few days ago, and then I saw all of the updated api discussion and decided to push pause...

**Comment by ljleb** [comment]
Whether we have tests or not to watch the code, I think this is a good idea. Should we create a dev branch + manual workflow to merge into main @Mikubill? So that we can merge PRs that look good into dev and merge commits that work well into main.

**Reply by Vespinian** [reply]
If you are going to do that, is there a way of telling on what commit hash of the webui the main was tested and seemed good with, in the rare cases where changes upstream might break something here.
